### PR TITLE
[Behat/app context] Check if current user is organizer #181

### LINKED ISSUE #181: [Behat] Event Application Context - regular user can create new events for organization
There's a false positive kind of bug in the EventApplicationContext, where a regular user, Jo Johnson, can create a new event for an organization.

### PR DESCRIPTION
**Pull request by HcsOmot**
There was a bug in the feature file for Event Application Context which
was a false positive - user who was not an organization organizer was
able to create a new event for organization.

Closes #181

### CODE REVIEW

**Comment by msvrtan** [comment]
Organizer check should be part of Handler, not Application context...

But for instance, it would make sense to put it in DomainContext on that step

**Comment by HcsOmot** [comment]
Seeing there is no handler for creating new events, should we merge this as is for now?

I can and the command and handler for creating events and update the relevant contexts in a separate PR.

Or I can just add the handler to this PR in additional commits.

**Comment by HcsOmot** [comment]
UPDATE:

I've added command & handler for creating new Events, and implemented `LocationRepository` based on how @msvrtan created `EventRepository` and `OrganizationRepository`.

I haven't changed anything in Behat context files just yet, because I'm not sure that `CreateEventHandler` is the place to perform the checks for current user's permissions (is the current user an organizer of some organization and can they create a new event).

I think this kind of check should be done somewhere else, but still not sure where and how.

Perhaps controller isn't such a bad place for this ATM? We haven't decided if this restriction should be part of the core business rules. If it is, we can always modify the command and handler to include the user who is trying to create the event and handle that in the Handler, however we see fit.

Please leave feedback on this, as we need to decide on where and how to implement these check before this gets merged to master.

@msvrtan @div3r @zgperak @flackovic @leovujanic

File: src/Event/Command/CreateEvent.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    private $eventDescription;
+
+    private $organizationId;
+
+    public function __construct(
+        EventId $eventId,
+        DateTime $dateTime,
+        LocationId $locationId,
+        string $argument1,
```
Comment (msvrtan) [inline src/Event/Command/CreateEvent.php]: Hm what?

**Comment by msvrtan** [comment]
I think handler is the best place to check if user is allowed to add an event.

Of course, there might be more places where we will need to add checks as well but handler is for sure a good one ...

File: src/Event/Command/CreateEvent.php (hunk 1 above)
Comment (HcsOmot) [reply src/Event/Command/CreateEvent.php]: Damn. Sorry, I was using memio spec-gen extension, totally missed these arguments.

**Comment by HcsOmot** [comment]
Ok, I'll add event creator to command.
:+1:

**Comment by HcsOmot** [comment]
Note to self:
We need negative scenario tests to prove users who aren't organizers can't create events.

**Comment by msvrtan** [comment]
@HcsOmot fire away!

**Pull request merged**